src/main/java/gov/usgs/wma/statistics/logic/StatisticsCalculator.java

```diff
@@ -86,10 +93,44 @@ public JsonData calculate(Specifier spec, List<S> samples)  {
 	public List<S> conditioning(Specifier spec, List<S> samples) {
 		removeNulls(samples, spec.getAgencyCd()+":"+spec.getSiteNo());
 		removeProvisional(samples, spec.toString());
+		checkAllDates(samples);
 		return samples;
 	}
 	
 	
+	public void checkAllDates(List<S> samples) {
+
+		// java does not have stream().forEachWithIndex()
+		String today = today();
+
+		for (int i=0; i<samples.size(); i++) {
+			S sample = samples.get(i);
+			String utc = sample.time;
+			String msg;
+			if ( isBlank(utc) ) {
+				msg = env.getError(ENV_INVALID_ROW_DATE_BLANK, i+1);
+				builder.error(msg);
+			} else if ( today.compareTo(utc) == -1 ) {
+				msg = env.getError(ENV_INVALID_ROW_DATE_FUTURE, i+1, utc);
+				builder.error(msg);
+			} else {
+				String fixed = fixMissingMonthAndDay(utc);
+				int delta = fixed.length() - utc.length();
+				sample.time = fixed;
+				if (delta <= 0) {
+					continue; // date is fine and was not fixed
+				} else if (delta <= 3) {
```

Comment:
Would the condition on line 120 and the condition on line 122 both be satisfied if the delta was less than or equal to 0?  Is that intentional?

Reply:
First, check if it is <= Zero and continue.
Now, it must be >= 1 but we want to handle the between 1 and 3 inclusive.

Reply:
Gotcha, that makes sense!